src/main/java/de/mediathekview/mserver/crawler/wdr/parser/WdrFilmDeserializer.java

```diff
@@ -0,0 +1,282 @@
+      aFilm.addAudioDescription(qualitiesEntry.getKey(), CrawlerTool.stringToFilmUrl(qualitiesEntry.getValue()));
+    }
+
+    videoUrls = aVideoInfo.getSignLanguageUrls();
+    for (final Map.Entry<Resolution, String> qualitiesEntry : videoUrls.entrySet()) {
+      aFilm.addSignLanguage(qualitiesEntry.getKey(), CrawlerTool.stringToFilmUrl(qualitiesEntry.getValue()));
+    }
+  }
+
+  private Optional<WdrVideoInfoDTO> parseUrls(final Document aDocument) {
+    final Optional<CrawlerUrlDTO> videoUrlDto = parseVideoLink(aDocument);
+    if (!videoUrlDto.isPresent()) {
+      return Optional.empty();
+    }
+    Optional<String> javaScriptContent = readContent(videoUrlDto.get().getUrl());
+    Optional<String> embeddedJson = extractJsonFromJavaScript(javaScriptContent);
+    if (!embeddedJson.isPresent()) {
+      return Optional.empty();
+    }
+      
+    Optional<WdrMediaDTO> mediaDto = gson.fromJson(embeddedJson.get(), OPTIONAL_WDRMEDIADTO_TYPE_TOKEN);
+    if (!mediaDto.isPresent()) {
+      return Optional.empty();
+    }
+    
+    WdrMediaDTO dto = mediaDto.get();
+    if (dto.getUrl().endsWith(".m3u8")) {
+      return parseM3U8(dto);
+    } else if(dto.getUrl().endsWith(".mp4")) {
+      return parseMP4Url(dto);
+    }
+    
+    return Optional.empty();
+  }
+  
+  private Optional<WdrVideoInfoDTO> parseM3U8(final WdrMediaDTO aMediaDto) {
+    Map<Resolution, String> urlMap = parseM3U8Url(aMediaDto.getUrl());
+    if (!urlMap.isEmpty()) {
+      WdrVideoInfoDTO videoInfoDto = new WdrVideoInfoDTO();
+      urlMap.forEach((key, value) -> videoInfoDto.putVideo(key, value));
+      
+      if (aMediaDto.getSubtitle().isPresent()) {
+        videoInfoDto.setSubtitleUrl(aMediaDto.getSubtitle().get());
+      }
+
+      if (aMediaDto.getAudioDescriptionUrl().isPresent()) {
+        Map<Resolution, String> adUrlMap = parseM3U8Url(aMediaDto.getAudioDescriptionUrl().get());
+        adUrlMap.forEach((key, value) -> videoInfoDto.putAudioDescription(key, value));
+      }
+
+      if (aMediaDto.getSignLanguageUrl().isPresent()) {
+        Map<Resolution, String> slUrlMap = parseM3U8Url(aMediaDto.getSignLanguageUrl().get());
+        slUrlMap.forEach((key, value) -> videoInfoDto.putSignLanguage(key, value));
+      }
+
+      return Optional.of(videoInfoDto);
+    }
+    
+    return Optional.empty();
+  }
+  
+  private Optional<WdrVideoInfoDTO> parseMP4Url(final WdrMediaDTO aMediaDto) {
+    WdrVideoInfoDTO dto = new WdrVideoInfoDTO();
+    dto.putVideo(Resolution.NORMAL, aMediaDto.getUrl());
+    
+    return Optional.of(dto);
+  }
+  
+  private Map<Resolution, String> parseM3U8Url(final String aUrl) {
+    Map<Resolution, String> urlMap = new HashMap<>();
+    
+    Optional<String> m3u8Content = readContent(aUrl);
+    if (!m3u8Content.isPresent()) {
+      return urlMap;
+    }
+    
+    M3U8Parser parser = new M3U8Parser();
+    List<M3U8Dto> m3u8Data = parser.parse(m3u8Content.get());
+
+    m3u8Data.forEach((entry) -> {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the parentheses around the "entry" parameter (sonar.java.source not set. Assuming 8 or greater.) [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1611)